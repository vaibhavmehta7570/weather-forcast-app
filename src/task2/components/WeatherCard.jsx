import React from 'react'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCaretUp, faCaretDown } from '@fortawesome/free-solid-svg-icons'
import weatherIcons from '../weatherIcons'

const WeatherCard = ({ weather }) => {
  const getIconClasses = () => {
    //importing class for given icon id from weatherIcons.js and adding the classes to weathericon tag
    let iconID = weather.iconID
    let classes = 'weathercard__icon wi wi-'
    let icon = weatherIcons[iconID].icon

    // If we are not in the ranges mentioned above, add a day/night prefix.
    icon =
      !(iconID > 699 && iconID < 800) && !(iconID > 899 && iconID < 1000)
        ? `day-${icon}`
        : icon
    return classes + icon
  }

  return (
    <article className="weathercard">
      <header className="weathercard__header">{weather.day}</header>
      <section className="weathercard__body">
        <section className="weathercard__body--info">
          <span className="weathercard__detail">{weather.city}</span>
          <span className="weathercard__detail">{weather.desc}</span>
        </section>
        <section className="weathercard__body--info">
          <div className="weathercard__temp">
            <span className="weathercard__maxMin">
              <span>
                {weather.temperatureMax}
                <span>&#176;</span>C
              </span>
              <FontAwesomeIcon className="faCaret faCaretUp" icon={faCaretUp} />
            </span>
            <span className="weathercard__maxMin">
              <span>
                {weather.temperatureMin}
                <span>&#176;</span>C{' '}
              </span>
              <FontAwesomeIcon
                className="faCaret faCaretDown"
                icon={faCaretDown}
              />
            </span>
          </div>
          <i className={getIconClasses()}></i>
        </section>
        <section className="weathercard__body--info">
          <span className="weathercard__detail weathercard__detail--thin">
            Humidity <br />
            {weather.humidity} %
          </span>
          <span className="weathercard__detail weathercard__detail--thin">
            Wind speed <br />
            {weather.windSpeed} knots
          </span>
        </section>
      </section>
    </article>
  )
}

export default WeatherCard
