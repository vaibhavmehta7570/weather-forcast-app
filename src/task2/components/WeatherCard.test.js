import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { configure } from 'enzyme'
import WeatherCard from './WeatherCard'
import renderer from 'react-test-renderer'
configure({ adapter: new Adapter() })

describe('Should test WeatherCard.jsx', () => {
  const mockValues = {
    id: 0,
    city: 'goa',
    day: '2021-01-28',
    desc: 'few clouds',
    humidity: 49,
    iconID: 801,
    temperatureMax: 29,
    temperatureMin: 20.2,
    windSpeed: 3.16
  }

  test('should render correctly weatherCard component', () => {
    const renderResponse = renderer
      .create(<WeatherCard weather={mockValues} />)
      .toJSON()
    expect(renderResponse).toMatchSnapshot()
  })

  test('should render correctly weatherCard component with 699 < iconID < 800', () => {
    const mockValues = {
      id: 0,
      city: 'goa',
      day: '2021-01-28',
      desc: 'few clouds',
      humidity: 49,
      iconID: 771,
      temperatureMax: 29,
      temperatureMin: 20.2,
      windSpeed: 3.16
    }
    const renderResponse = renderer
      .create(<WeatherCard weather={mockValues} />)
      .toJSON()
    expect(renderResponse).toMatchSnapshot()
  })

  test('should render correctly weatherCard component with 899 < iconID <1000', () => {
    const mockValues = {
      id: 0,
      city: 'goa',
      day: '2021-01-28',
      desc: 'few clouds',
      humidity: 49,
      iconID: 904,
      temperatureMax: 29,
      temperatureMin: 20.2,
      windSpeed: 3.16
    }
    const renderResponse = renderer
      .create(<WeatherCard weather={mockValues} />)
      .toJSON()
    expect(renderResponse).toMatchSnapshot()
  })
})
