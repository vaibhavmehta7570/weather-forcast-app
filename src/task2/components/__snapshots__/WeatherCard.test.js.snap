// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should test WeatherCard.jsx should render correctly weatherCard component 1`] = `
<article
  className="weathercard"
>
  <header
    className="weathercard__header"
  >
    2021-01-28
  </header>
  <section
    className="weathercard__body"
  >
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail"
      >
        goa
      </span>
      <span
        className="weathercard__detail"
      >
        few clouds
      </span>
    </section>
    <section
      className="weathercard__body--info"
    >
      <div
        className="weathercard__temp"
      >
        <span
          className="weathercard__maxMin"
        >
          <span>
            29
            <span>
              °
            </span>
            C
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-up fa-w-10 faCaret faCaretUp"
            data-icon="caret-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="weathercard__maxMin"
        >
          <span>
            20.2
            <span>
              °
            </span>
            C
             
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down fa-w-10 faCaret faCaretDown"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <i
        className="weathercard__icon wi wi-day-cloudy"
      />
    </section>
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Humidity 
        <br />
        49
         %
      </span>
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Wind speed 
        <br />
        3.16
         knots
      </span>
    </section>
  </section>
</article>
`;

exports[`Should test WeatherCard.jsx should render correctly weatherCard component with 699 < iconID < 800 1`] = `
<article
  className="weathercard"
>
  <header
    className="weathercard__header"
  >
    2021-01-28
  </header>
  <section
    className="weathercard__body"
  >
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail"
      >
        goa
      </span>
      <span
        className="weathercard__detail"
      >
        few clouds
      </span>
    </section>
    <section
      className="weathercard__body--info"
    >
      <div
        className="weathercard__temp"
      >
        <span
          className="weathercard__maxMin"
        >
          <span>
            29
            <span>
              °
            </span>
            C
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-up fa-w-10 faCaret faCaretUp"
            data-icon="caret-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="weathercard__maxMin"
        >
          <span>
            20.2
            <span>
              °
            </span>
            C
             
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down fa-w-10 faCaret faCaretDown"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <i
        className="weathercard__icon wi wi-day-windy"
      />
    </section>
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Humidity 
        <br />
        49
         %
      </span>
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Wind speed 
        <br />
        3.16
         knots
      </span>
    </section>
  </section>
</article>
`;

exports[`Should test WeatherCard.jsx should render correctly weatherCard component with 899 < iconID <1000 1`] = `
<article
  className="weathercard"
>
  <header
    className="weathercard__header"
  >
    2021-01-28
  </header>
  <section
    className="weathercard__body"
  >
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail"
      >
        goa
      </span>
      <span
        className="weathercard__detail"
      >
        few clouds
      </span>
    </section>
    <section
      className="weathercard__body--info"
    >
      <div
        className="weathercard__temp"
      >
        <span
          className="weathercard__maxMin"
        >
          <span>
            29
            <span>
              °
            </span>
            C
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-up fa-w-10 faCaret faCaretUp"
            data-icon="caret-up"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
        <span
          className="weathercard__maxMin"
        >
          <span>
            20.2
            <span>
              °
            </span>
            C
             
          </span>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-caret-down fa-w-10 faCaret faCaretDown"
            data-icon="caret-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </span>
      </div>
      <i
        className="weathercard__icon wi wi-hot"
      />
    </section>
    <section
      className="weathercard__body--info"
    >
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Humidity 
        <br />
        49
         %
      </span>
      <span
        className="weathercard__detail weathercard__detail--thin"
      >
        Wind speed 
        <br />
        3.16
         knots
      </span>
    </section>
  </section>
</article>
`;
